ttnn/cpp/ttnn/tensor/types.hpp

```diff
@@ -404,6 +412,11 @@ struct BorrowedStorage {
 
     static constexpr auto attribute_names = std::forward_as_tuple();
     const auto attribute_values() const { return std::forward_as_tuple(); }
+
+    inline bool is_allocated() const {
```

Comment:
This method can be `static constexpr bool is_allocated()` since it doesn't require the `this` pointer. Static methods can also syntactically be invoked with `.` on instances like member methods, so there's no changes necessary at the call sites to accommodate this change.

Reply:
It is a so small issue. Also it removes attention from the real fix: million of mutex locks in the Multi* storages in one is_allocated call. Plus obvious  multithreading bug.